### ISSUE TITLE
Creating a mock with coAnswers that contain certain suspending functions causes tests to hang

### ISSUE THREAD

**Issue by TAGC** [mockk/mockk#171]
### Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Expected Behavior

Given a unit test where I set up a mock like this:

```kotlin
val executionCompletionSource = CompletableDeferred<Nothing>()
suspend fun task(): Unit = executionCompletionSource.await()
val mock = mockk<Executable> { coEvery { execute() } coAnswers { task() } }
// ...
val execution = launch { mock.execute() } 
executionCompletionSource.cancel()
```

I expect that the unit test eventually completes.

### Current Behavior

A unit test with this code ends up hanging indefinitely, even though an analogous non-mock version of this procedure does not hang.

#### Steps to Reproduce

1. Create a Gradle project with these dependencies:

    ```gradle
    implementation(kotlin("stdlib-jdk8"))
    implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.0.0")
    implementation("io.mockk:mockk:1.8.10.kotlin13")
    ```

2.  Create a file within the project with this content:

    ```kotlin
    package experiment

    import io.mockk.coEvery
    import io.mockk.mockk
    import kotlinx.coroutines.CompletableDeferred
    import kotlinx.coroutines.delay
    import kotlinx.coroutines.launch
    import kotlinx.coroutines.runBlocking
    import kotlin.system.measureTimeMillis

    interface Executable {
        suspend fun execute()
    }

    fun main() {
        val executionCompletionSource = CompletableDeferred<Nothing>()
        suspend fun task(): Unit = executionCompletionSource.await()
        val mock = mockk<Executable> { coEvery { execute() } coAnswers { task() } }

        runBlocking {
            val execution = launch { mock.execute() } // This blocks the test indefinitely
            //val execution = launch { task() } // This works fine (~110-120 ms)
            measureTimeMillis {
                delay(100)
                executionCompletionSource.cancel()
                execution.join()
            }.also { elapsed -> println("Elapsed in $elapsed ms") }
        }
    }
    ```

3. Run that code, verify that it hangs indefinitely.
4. Comment out the `val execution = launch { mock.execute() } ` line and uncomment the line beneath it
5. Try running the code again, verifying that it no longer hangs.

#### Context

* MockK version: 1.8.10.kotlin13
* OS: Windows 10 (64-bit)
* Kotlin version: 1.3.0
* JDK version: 1.8.0_191 
* JUnit version: 5
* Type of test: unit test

**Comment by TAGC** [comment]
[Relevant SO question originally opened about this](https://stackoverflow.com/questions/53223093/how-can-i-mock-execution-of-coroutines-using-mockk)

**Comment by oleksiyp** [comment]
Simplest workaround is:
```
val execution = launch(start = CoroutineStart.LAZY) { mock.execute() } // This blocks the test indefinitely
```

**Comment by oleksiyp** [comment]
The problem is coroutines a lot more complex than it can be on a first sight. 
When I was writing the code for coroutines I didn't know all that stuff under the hood.
MockK just catches calls and inside do `runBlocking`

The problem here is that by default `launch` doesn't spawn anything. 
It just executes all in same execution thread. 
When suspension occurs callback will be resumed on other thread.

MockK catches the call to `execute` and inside just do `runBlocking`, so the main thread is not suspending. 

MockK should be transparent layer for such calls. This will require some investigation and rework.

**Comment by TAGC** [comment]
I can confirm that works. Interesting, if you replace the `delay(100)` line with `withTimeoutOrNull(100) { execution.join() } shouldBe null`, it hangs indefinitely again.

**Comment by TAGC** [comment]
Launching with `CoroutineStart.LAZY` isn't a suitable workaround for my case anyway, as it causes my test to pass even when it should fail. I'll create a handwritten mock instead for the time being.

**Comment by oleksiyp** [comment]
New release with fixes is out `v1.8.13` or `v1.8.13.kotlin13`. Please verify

**Comment by TAGC** [comment]
Yep, with v1.8.13 the test passes when it should and fails when it should. Cheers 👍
